services/workflows-service/src/storage/get-file-storage-manager.ts

```diff
@@ -15,15 +15,23 @@ export const S3StorageEnvSchema = z.object({
   AWS_S3_BUCKET_KEY: z.string(),
 });
 
-const __generateAwsConfig = (processEnv: NodeJS.ProcessEnv): S3ClientConfig => {
-  const { AWS_REGION, AWS_S3_BUCKET_KEY, AWS_S3_BUCKET_SECRET } =
-    S3StorageEnvSchema.parse(processEnv);
+export const generateAwsConfig = (processEnv: NodeJS.ProcessEnv, prefixConfigName?: string): S3ClientConfig => {
+  const { AWS_REGION, AWS_S3_BUCKET_KEY, AWS_S3_BUCKET_SECRET } = S3StorageEnvSchema.parse(processEnv);
```

Comment:
I wouldve expect to have S3-specific stuff in some s3 adapter, so someone can easily duplicate it and add another provider

Reply:
I need suggestions regarding how should I approach this issue